File: programmers/C++/Level1/크레인 인형뽑기 게임.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<vector<int>> board, vector<int> moves) {

    int answer = 0;
    vector<int> basket;

    for (int i = 0; i < moves.size(); i++)
    {
        int grab = moves[i] - 1;
        for (int j = 0; j < board.size(); j++)
        {
            int value = board[j][grab];
            if (value != 0)  // board에서 인형 발견 시
            {
                board[j][grab] = 0;
                basket.push_back(value);

                int len = basket.size();
                if (len > 1 && basket[len - 2] == basket[len - 1])  // 바구니에 같은 인형이 만난 경우
                {
                    basket.pop_back();
                    basket.pop_back();
                    answer += 2;
                }

                break;
            }
        }
    }

    return answer;
}

File: programmers-coding-test/Python/Level1/크레인 인형뽑기 게임.py
def solution(board, moves):
    popped, basket = 0, []
    for m in moves:
        for i in range(len(board)):
            if board[i][m - 1] != 0:
                basket.append(board[i][m - 1])
                if len(basket) > 1 and basket[-1] == basket[-2]:
                    basket.pop()
                    basket.pop()
                    popped += 2
                board[i][m - 1] = 0
                break
    return popped
